### ISSUE TITLE
Build for windows throws javabridge exception

### ISSUE THREAD

**Issue by daviddao** [CellProfiler/CellProfiler-Analyst#132]
I managed to build and call CPA as app on windows. But when I call the app with a properties file, javabridge throws following exception: 

```
C:\Users\daviddao\Desktop\CellProfiler-Analyst-master\dist\library.zip\matplotlib\__init__.py:872: UserWarning: axes.color_cycle is deprecated and replaced with axes.prop_cycle; please use the latter.
INFO:root:No PIL.ArgImagePlugin
WARNING:root:PROPERTIES WARNING (image_channel_paths): This field name has been deprecated, use "image_path_cols" instead.
WARNING:root:PROPERTIES WARNING (image_channel_files): This field name has been deprecated, use "image_file_cols" instead.
WARNING:root:PROPERTIES WARNING (image_channel_names): This field name has been deprecated, use "image_names" instead.
WARNING:root:PROPERTIES WARNING (channels_per_image): No value(s) specified. CPA will assume 1 channel per image.
INFO:root:PROPERTIES: Using default image_buffer_size=1
INFO:root:PROPERTIES: Using default tile_buffer_size=1
INFO:root:PROPERTIES: Training set found at "C:\Users\daviddao\Desktop\cpa_2.0_example\cpa_example\MyTrainingSet.txt"
WARNING:root:PROPERTIES WARNING (well_format): Field was not defined, using default format of "A01".
INFO:root:[MainThread] Connecting to the database...
INFO:root:[MainThread] SQLite file: C:\Users\daviddao\CPA\CPA_DB_d01b8a5b47a95a1bb1b469cdcf2a804c.db
Failed to get class org/mozilla/javascript/Context
Traceback (most recent call last):
  File "CellProfiler-Analyst.py", line 543, in <module>
  File "CellProfiler-Analyst.py", line 464, in Start
  File "javabridge\jutil.pyc", line 687, in activate_awt
  File "javabridge\jutil.pyc", line 334, in run_script
  File "javabridge\jutil.pyc", line 902, in static_call
  File "javabridge\jutil.pyc", line 876, in make_static_call
javabridge.jutil.JavaException: org/mozilla/javascript/Context
```

I have the latest JDK1.8 installed on Windows Vista (Broad PC) and the JAVA_HOME variable is set correctly. In Powershell, this static call works in the python REPL: 

```
javabridge.static_call('org/mozilla/javascript/Context','enter','()Lorg/mozilla/javascript/Context;')
```

* py2exe file: https://github.com/CellProfiler/CellProfiler-Analyst/blob/master/setup_py2exe.py
* supposed error line: https://github.com/CellProfiler/CellProfiler-Analyst/blob/master/CellProfiler-Analyst.py#L464

@LeeKamentsky Any thoughts about it?

**Comment by LeeKamentsky** [comment]
I think the JVM is going at that point. I can't see the Java exception in your log (there would have been one if it was CellProfiler), but in that context, it's pretty obvious that you don't have the Javabridge jar files on your class path. I had a fix in December that post-dates the 1.0.11 Javabridge release - you might first try a pip install of the master branch and see if that helps.

Also, the line

    javabridge.activate_awt()

(https://github.com/CellProfiler/CellProfiler-Analyst/blob/master/CellProfiler-Analyst.py#L464) must be removed. It's totally inappropriate - asking the javabridge to start the AWT event loop. I'm surprised it works on the Mac.

You can get the Java class path after you start the JVM (and then put it in your logger output in order to check if you're getting the jars correctly) like this:

    import javabridge
    import os
    class_path = javabridge.JClassWrapper('java.lang.System').getProperty('java.class.path')
    print "\n".join([
        '%s %s' % (jarpath, 'exists' if os.path.exists(jarpath) else 'missing')
         for jarpath in class_path.split(os.pathsep)])

my output:

    c:\Python27\lib\site-packages\javabridge-1.0.11_21_g881902f-py2.7-win-amd64.egg\javabridge\jars\rhino-1.7R4.jar exists
    c:\Python27\lib\site-packages\javabridge-1.0.11_21_g881902f-py2.7-win-amd64.egg\javabridge\jars\runnablequeue.jar exists
    c:\Python27\lib\site-packages\javabridge-1.0.11_21_g881902f-py2.7-win-amd64.egg\javabridge\jars\cpython.jar exists
    c:\users\leek\cpdev\python-bioformats\python-bioformats\bioformats\jars\loci_tools.jar exists

**Comment by daviddao** [comment]
Thanks! I will give it a try!

**Comment by daviddao** [comment]
@LeeKamentsky : You are right, somehow the javabridge JARs are missing.

Hence, I installed the master-branch of python-javabridge in PowerShell directly via git:

```
C:\Program Files\Git\cmd>python -m pip install git+https://github.com/CellProfiler/python-javabridge
C:\Python27\lib\site-packages\pip\pep425tags.py:89: RuntimeWarning: Config variable 'Py_DEBUG' is unset, Python ABI tag
may be incorrect
  warn=(impl == 'cp')):
C:\Python27\lib\site-packages\pip\pep425tags.py:93: RuntimeWarning: Config variable 'WITH_PYMALLOC' is unset, Python ABI
 tag may be incorrect
  warn=(impl == 'cp')):
C:\Python27\lib\site-packages\pip\pep425tags.py:99: RuntimeWarning: Config variable 'Py_UNICODE_SIZE' is unset, Python A
BI tag may be incorrect
  sys.version_info < (3, 3))) \
Collecting git+https://github.com/CellProfiler/python-javabridge
  Cloning https://github.com/CellProfiler/python-javabridge to c:\users\daviddao\appdata\local\temp\pip-w7kftu-build
Requirement already satisfied (use --upgrade to upgrade): numpy in c:\python27\lib\site-packages (from javabridge==0.0.0
)
Installing collected packages: javabridge
  Found existing installation: javabridge 1.0.11
    Uninstalling javabridge-1.0.11:
      Successfully uninstalled javabridge-1.0.11
  Running setup.py install for javabridge ... done
Successfully installed javabridge-0.0.0
```

After building and commenting out:

```
javabridge.activate_awt()
```

I still end up with the missing JARs

```
C:\Users\daviddao\Desktop\CellProfiler-Analyst-master\dist\library.zip\javabridge\jars\rhino-1.7R4.jar missing
C:\Users\daviddao\Desktop\CellProfiler-Analyst-master\dist\library.zip\javabridge\jars\runnablequeue.jar missing
C:\Users\daviddao\Desktop\CellProfiler-Analyst-master\dist\library.zip\javabridge\jars\cpython.jar missing
C:\Users\daviddao\Desktop\CellProfiler-Analyst-master\dist\library.zip\bioformats\jars\loci_tools.jar missing
```

Did I installed javabridge correctly?

**Comment by daviddao** [comment]
Strangely, if run on the command line, CPA has no problems finding the JARs:

```
C:\Python27\lib\site-packages\javabridge\jars\rhino-1.7R4.jar exists
C:\Python27\lib\site-packages\javabridge\jars\runnablequeue.jar exists
C:\Python27\lib\site-packages\javabridge\jars\cpython.jar exists
C:\Python27\lib\site-packages\bioformats\jars\loci_tools.jar exists
```

**Comment by LeeKamentsky** [comment]
oh sorry about the late reply.. You are bundling javabridge and bioformats into the (stupid) zip file of dependencies and if you look at the path, it goes straight into the zipfile and that will never work. The solution is probably to set the "skip_archive" flag in your py2exe options:

    options={
        'py2exe': {
            'skip_archive':1,
         ... } }

**Comment by daviddao** [comment]
Solved with ae4086c :tada: :relieved:
